src/ui/utils/testRuns.ts

```diff
@@ -0,0 +1,71 @@
+import orderBy from "lodash/orderBy";
+
+import { Recording } from "shared/graphql/types";
+
+function testPassed(recording: Recording) {
+  return recording.metadata?.test?.result === "passed";
+}
+
+function testFailed(recording: Recording) {
+  return (
+    recording.metadata?.test?.result &&
+    ["failed", "timedOut"].includes(recording.metadata?.test?.result)
+  );
+}
+export function groupRecordings(recordings: Recording[] | undefined) {
+  const passedRecordings: Recording[] = [];
+  const failedRecordings: Recording[] = [];
+  const flakyRecordings: Recording[] = [];
+  const sortedRecordings = orderBy(recordings, "date", "desc");
+
+  const flakySet = new Set<string>();
+
+  const fileCounts = sortedRecordings.reduce((acc, recording) => {
+    const file = recording.metadata?.test?.file;
+    if (!file) {
+      return acc;
+    }
+    if (!acc[file]) {
+      acc[file] = 0;
+    }
+    acc[file]++;
+    return acc;
+  }, {} as Record<string, number>);
```

Comment:
Performance nit: If you changed the structure of this from a file-to-count map to a file-to-array-of-recordings map, you could avoid having to loop over the entire recordings array to filter by file on line 42 (and just iterate over the map values instead).

Reply:
Good call. I think the code is simpler too